src/core/thread/mle_router.cpp

```diff
@@ -3318,7 +3318,9 @@ void MleRouter::RemoveRouterLink(Router &aRouter)
 
 void MleRouter::RemoveNeighbor(Neighbor &aNeighbor)
 {
-    if (&aNeighbor == &mParent)
+    VerifyOrExit(!aNeighbor.IsStateInvalid());
```

Comment:
It is possible that `aNeighbor` is invalid here when called from `MleRouter::HandleAdvertisement`: 
https://github.com/openthread/openthread/blob/af4837d122c9da0eb6dcb9f2a1b19b1e37b738cc/src/core/thread/mle_router.cpp#L1425-L1429

I think it might be a side effect of the fix of this PR in `Mle::HandleLeaderData`.